presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/TemporalFunction.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.storage.organization;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.Type;
+
+public interface TemporalFunction
+{
+    int getDay(ConnectorSession session, Block block, int position);
+
+    int getDay(Object min, Object max);
```

Comment:
Why are parameters have type `Object` and not more specific type? Also, it's not clear from the method signature what this function is intended to do, so think of a better name for it and potentially a javadoc description.